.github/workflows/code-engine-cleanup.yml

```diff
@@ -30,22 +34,28 @@ on:
 jobs:
   clean:
     runs-on: ubuntu-latest
-    env:
-      PR_TAG: pr-${{ github.event.pull_request.number }}
 
     steps:
+      - name: Set preview tag using input
+        if: "${{ inputs.preview_tag != '' }}"
+        run: echo "PREVIEW_TAG=${{ inputs.preview_tag }}" >> $GITHUB_ENV
+
+      - name: Set preview tag using PR number
+        if: "${{ inputs.preview_tag == '' }}"
+        run: echo "PREVIEW_TAG=pr-${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+
       - name: Create APP_NAME and DOCKER_IMAGE_TAG env variables
         run: |
-          echo "APP_NAME=${{ inputs.docker_image_name }}-${{ env.PR_TAG }}" >> $GITHUB_ENV
-          echo "DOCKER_IMAGE_TAG=icr.io/${{ inputs.container_registry_namespace }}/${{ inputs.docker_image_name }}:${{ env.PR_TAG }}" >> $GITHUB_ENV
+          echo "APP_NAME=${{ inputs.docker_image_name }}-${{ env.PREVIEW_TAG }}" >> $GITHUB_ENV
+          echo "DOCKER_IMAGE_TAG=icr.io/${{ inputs.container_registry_namespace }}/${{ inputs.docker_image_name }}:${{ env.PREVIEW_TAG }}" >> $GITHUB_ENV
 
       - name: GH deployment status (deactivate)
         id: gh_deployment
         uses: bobheadxi/deployments@v0.6.2
         with:
           step: deactivate-env
           token: ${{ github.token }}
-          env: preview-${{ env.PR_TAG }}
+          env: preview-${{ env.PREVIEW_TAG }}
```

Comment:
We should allow override this? To have the three envs:
- Preview
- Staging
- Production

Reply:
thank you.... good point! 
done